pytorch_lightning/strategies/ddp.py

```diff
@@ -144,18 +135,19 @@ def distributed_sampler_kwargs(self):
     def _is_single_process_single_device(self) -> bool:
         return True
 
-    def setup_environment(self) -> None:
-        # start the other scripts
-        if not self.cluster_environment.creates_processes_externally:
-            self._call_children_scripts()
+    def configure_multi_process_launcher(self):
```

Comment:
I'd like to avoid this, need to brainstorm.

Reply:
Yes, its annoying, I know. But we can't make the decision for it during init. One workaround can be to construct it with strategy and make the decision whether to create children scripts inside Launcher.launch.

Reply:
Do you need to make a decision at init? It could be done when the launch is triggered no?

Reply:
yes, but for that, we have to put the logic inside the launcher and construct `strategy` inside that instead of num_processes, nodes and cluster environment. Also if the decision is made inside and for the cases where processes are created externally, subprocess scripts won't be created and calling it `SubprocessScriptLauncher` might not be completely correct.